hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -116,6 +116,13 @@ private OzoneClient getClient(OzoneConfiguration config) throws IOException {
     }
   }
 
+  // ONLY validate aws access id when needed.
+  private void validateAccessId(String awsAccessId) throws Exception {
+    if (awsAccessId == null || awsAccessId.equals("")) {
+      throw S3_AUTHINFO_CREATION_ERROR;
```

Comment:
https://github.com/apache/hadoop-ozone/pull/1110#issuecomment-648405855

When S3 misses the auth header error returned is AuthorizationHeaderMalformedHeader.

Question: After this fix, will it show this error when the auth header is missing or it will be just logged in S3G logs?

Reply:
@bharatviswa504 Right now, it catching all exceptions and just print out logs. Do we want to throw OS3Exception?

Reply:
Yes, When I tried AWS S3 end point it shows AuthorizationHeaderMalformed to user.
IMHO S3G needs to mimic AWS S3 behavior, so S3 tools can handle S3G.

Reply:
@bharatviswa504 I agree. But it will involve discussion over our internal errors mapping to s3 error codes, which could be a larger discussion. This patch is to have a hot fix for AWSSignatureProcessor init process to avoid NPE. Shall I create a separate JIRA to track handling s3g errors?

Reply:
fine with me if we want to fix the error in a new Jira.
Here we need to map this missing auth header to AuthorizationHeaderMalformed. We should mimic AWS S3 behavior like any other error like BucketNotFound etc.,

Reply:
During initial S3G development, we have followed this error list, and also for API we have tried with S3 to see the behavior and also documentation for that API.

https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList

See if this helps here.

Reply:
Open https://issues.apache.org/jira/browse/HDDS-4361 to track proper s3 errors back to bad requests.

Reply:
@bharatviswa504 I think a correct handling for ozone internal errors and s3 native errors is very important and would involve more in-depth changes between Ozone client and s3g. HDDS-4361 will be an umbrella JIRA to track the corrections we wanna make.